Documentation/proposals/202304-silences.md

```diff
@@ -0,0 +1,112 @@
+# Silence CRD
+
+* Owners:
+  * [mcbenjemaa](https://github.com/mcbenjemaa)
+* Related Tickets:
+  * [#5452](https://github.com/prometheus-operator/prometheus-operator/issues/5452)
+  * [#2398](https://github.com/prometheus-operator/prometheus-operator/issues/2398)
+* Other docs:
+  * n/a
+
+This document describes the creation of `Silence` Custom Resource Definition that defines Alertmanager silences
+configurations in the Kubernetes way.
+
+# Why
+
+prometheus-operator doesn't have a way to automate the management of Alertmanager silences. Many users have either been using some internal scripts
+or an additional operator that does the job e.g. [silence-operator](https://github.com/giantswarm/silence-operator).
+
+* Users who are using CI/CD jobs to manage silences have reported that this is really cumbersome.
+* Users that use scripts are not in better shape obviously.
+* Users that use a standalone operator that exposes a Silence CRD are in a better situation, some folks have reported that they are using GitOps
+  to fully manage the life cycle of Silences Custom resources, Nevertheless, this comes with a drawback
+  as the team must be able to manage a new component in their stack.
+
+Additionally, having a new component in the stack and keeping it maintained is not always ideal (said the folks at [Giant Swarm](https://giantswarm.io) the owners of [silence-operator](https://github.com/giantswarm/silence-operator)),
+because that would require us to fully manage this component as of any managed app (monitoring, logging, alerting etc..).
+Having the Silence CRD as part of prometheus-operator provides users with a better experience: there is no need to install another operator or build a system on top of it to provide that functionality.
+
+By Adding support for `Silence` CRD in the prometheus-operator, this will make users more flexible in terms of choosing the tool
+to deploy their silences and will free others from managing a standalone component within the stack.
```

Comment:
```suggestion
Adding support for `Silence` CRD in the Prometheus-operator will provide more flexibility to users in terms of choosing the tool to deploy their silences and will free users of Giantswarm's silence operator from managing a standalone component within the stack.
```

I'm not sure about this suggestion, to be honest 😅 

"Free others" sounds too vague, but being very direct that this is helping users of Giantswarm home-grown solution sounds like an attack, which is not my intention 😬

Reply:
Yeah,

Reply:
@theo

Reply:
I think one of the idea here is that new users would be able to use this operator without having to think how to manage their silences and existing users would be able to use it but they could also use their existing mechanism if they wish

Reply:
So @QuentinBisson what do you suggest to keep here?

Reply:
I think Arthur suggestions make sense but maybe we do not need to talk specifically about the silence operator:

Adding support for `Silence` CRD in the Prometheus-operator will provide more flexibility to users in terms of choosing the tool to deploy their silences and will free users from managing a standalone component within the stack. 

That would work for both the silence operator or CI jobs or other tools that creative people came up with

Reply:
this is done